### PR TITLE
Sda design

### PR DESCRIPTION
**Pull request by brianray**

### CODE REVIEW

**Comment by truongc2** [comment]
To Do:

- [ ] Update document to match [template](https://github.com/data-describe/data-describe/blob/master/docs/designs/TEMPLATE.md)

What additional features do we need support for?
- Functionalities already supported: Data summary, scatterplots, feature importance (permutation importance), Time series
- Requested Features: [3D surface spectrogram](https://en.wikipedia.org/wiki/Spectrogram#:~:text=A%20spectrogram%20is%20a%20visual,they%20may%20be%20called%20waterfalls) and outlier detection

**Review by ankur-srivastava1 — APPROVED** [review]
This looks good!

File: docs/designs/anomaly_detection.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+* Interactive plots for selecting time window.
+
+## Design
+There are multiple design aspects to be considered when creating this functionality.
+1. Supervised: Partitioning the data into train and test sets to calculate the confidence bands and error rates.
+2. Unsupervised: Fitting a model on the entire data, i.e. HDBSCAN and isolation forests
+3. Statistical methods: i.e. standard deviation from rolling mean and SH-ESD.
+```python
+import data_describe as dd
+dd.anomaly_detection(df) #Autodetect datatype for classification, regression, timeseries
```
Comment (haishiro) [inline docs/designs/anomaly_detection.md:25]: Can/will this share an API with general anomaly detection that isn't time series? How would the API change for one "mode" versus another? What things are configurable?

File: docs/designs/anomaly_detection.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+* Creating notebooks for specific use cases i.e. sensor data, accelerometers, manufacturing.
+## UI or API
+* 2-D plot of time series data with confidence band. 
+* Use markers/colors to highlight anomalies.
+* Interactive plots for selecting time window.
+
+## Design
+There are multiple design aspects to be considered when creating this functionality.
+1. Supervised: Partitioning the data into train and test sets to calculate the confidence bands and error rates.
+2. Unsupervised: Fitting a model on the entire data, i.e. HDBSCAN and isolation forests
```
Comment (haishiro) [inline docs/designs/anomaly_detection.md:21]: Consider adding autoencoder as well - I believe HDBSCAN and isolation forests are more suited to non-time series

File: docs/designs/anomaly_detection.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+3. Statistical methods: i.e. standard deviation from rolling mean and SH-ESD.
```
Comment (haishiro) [inline docs/designs/anomaly_detection.md:22]: Consider looking at statistical process control - e.g. the Western Electric rules

**Review by haishiro — APPROVED** [review]
Approved, but added some comments for additional thinking / research that should be done before diving into code.

**Pull request merged**